test/integration/associations/has-many.test.js

```diff
@@ -2607,7 +2607,7 @@ describe(Support.getTestDialectTeaser('HasMany'), function() {
         User.hasMany(Tasks[dataType], { foreignKey: 'userId', keyType: dataType, constraints: false });
 
         return Tasks[dataType].sync({ force: true }).then(function() {
-          expect(Tasks[dataType].rawAttributes.userId.type.toString()).to.equal(dataType.toString());
+          expect(Tasks[dataType].rawAttributes.userId.type instanceof dataType).to.be.ok;
```

Comment:
`expect(Tasks[dataType].rawAttributes.userId.type instanceof dataType).to.be.an.instanceof(dataType);`